extensions/amp-story/1.0/media-performance-metrics-service.js

```diff
@@ -0,0 +1,413 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  MEDIA_LOAD_FAILURE_SRC_PROPERTY,
+  listen,
+} from '../../../src/event-helper';
+import {Services} from '../../../src/services';
+import {dev} from '../../../src/log';
+import {lastChildElement} from '../../../src/dom';
+import {map} from '../../../src/utils/object';
+import {registerServiceBuilder} from '../../../src/service';
+
+/**
+ * Media status.
+ * @enum
+ */
+const Status = {
+  ERRORED: 0,
+  PAUSED: 1,
+  PLAYING: 2,
+  WAITING: 3,
+};
+
+/**
+ * @typedef {{
+ *   start: number,
+ *   playing: number,
+ *   waiting: number,
+ * }}
+ */
+let TimeStampsDef;
+
+/**
+ * @typedef {{
+ *   error: ?number,
+ *   jointLatency: number,
+ *   rebuffers: number,
+ *   rebufferTime: number,
+ *   watchTime: number
+ * }}
+ */
+let MetricsDef;
+
+/**
+ * @typedef {{
+ *   media: !HTMLMediaElement,
+ *   status: number,
+ *   unlisteners: !Array<!UnlistenDef>,
+ *   timeStamps: !TimeStampsDef,
+ *   metrics: !MetricsDef
+ * }}
+ */
+let MediaEntryDef;
+
+/** @type {string} */
+const ID_PROPERTY = '__AMP_MEDIA_PERFORMANCE_METRICS_ID';
+
+/** @type {number} */
+const MINIMUM_TIME_THRESHOLD_MS = 1000;
+
+/** @type {number} */
+const REBUFFER_THRESHOLD_MS = 250;
```

Comment:
Is this standardized somewhere?  Is this a product call?

If not, I wonder if it should be drastically less (like, ~17ms e.g. 60 FPS).  If the experience had tons of 250ms rebuffers, we would want to know